### PR TITLE
Function to compare nodeinfos to find similar nodegroups

### PR DESCRIPTION
**Pull request by MaciekPytel**
Ref: #47

### CODE REVIEW

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+const (
+	// MaxAllocatableDifferencePercent describes how Node.Status.Allocatable can differ between
+	// groups in the same NodeGroupSet
+	MaxAllocatableDifferencePercent = 0.05
```
Comment (mwielgus) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: is it 5% or 0.05% ?

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	// MaxFreeDifferencePercent describes how free resources (allocatable - daemon and system pods)
+	// can differ between groups in the same NodeGroupSet
+	MaxFreeDifferencePercent = 0.05
```
Comment (mwielgus) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: Same here.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+)
+
+func compareResourceMapsWithTolerance(resources map[apiv1.ResourceName][]resource.Quantity,
```
Comment (mwielgus) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: why don't we pass 2 maps?

Comment (MaciekPytel) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: It's a private helper function, so I just passed the data structure I already had. As to why not use 2 maps in the first place - I don't think it matters much for temporary datastructure, that is never passed outside and this was just more convenient to create with a loop. I can refactor it if you think it's a bad choice.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go (hunk 1 above)
Comment (MaciekPytel) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: 5%, will rename the const.

Comment (MaciekPytel) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: done

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go (hunk 2 above)
Comment (MaciekPytel) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: done

**Comment by mwielgus** [comment]
lgtm

**Pull request merged**